ir/expression.def

```diff
@@ -321,7 +321,7 @@ class TypeNameExpression : Expression {
     dbprint{ out << typeName; }
     toString { return typeName->toString(); }
     validate { BUG_CHECK(typeName->is<Type_Name>() || typeName->is<Type_Specialized>(),
-                         "%1 unexpected type in TypeNameExpression", typeName); }
+                         "%1% unexpected type in TypeNameExpression", typeName); }
```

Comment:
the sooner we migrate out of boost::format, the better. The format strings should be checked statically at compile time as much as possible :)

Reply:
So we should use C `printf` instead so we can use `__attribute__((__format__` to get compile time checking?  I'm not aware of any way to get the C++ formatting stuff (either boost or std::fmt) to check things at compile time.

Reply:
abseil does this. fmtlib does as well. Actually, the places where we do use abseil formatting routines already perform compile-time checking (e.g. `SourceCodeBuilder::appendFormat`).